logstash-core/lib/logstash/api/commands/hot_threads_reporter.rb

```diff
@@ -3,12 +3,16 @@
 
 class HotThreadsReport
   STRING_SEPARATOR_LENGTH = 80.freeze
-  HOT_THREADS_STACK_TRACES_SIZE_DEFAULT = 10.freeze
+  HOT_THREADS_STACK_TRACES_SIZE_DEFAULT = 50.freeze
 
   def initialize(cmd, options)
     @cmd = cmd
     filter = { :stacktrace_size => options.fetch(:stacktrace_size, HOT_THREADS_STACK_TRACES_SIZE_DEFAULT) }
-    @thread_dump = ::LogStash::Util::ThreadDump.new(options.merge(:dump => ThreadsReport.generate(filter)))
+    filter2 = Hash.new
+    filter.each do |k,v|
+      filter2.store(k.to_s, v.to_s)
+    end
+    @thread_dump = ::LogStash::Util::ThreadDump.new(options.merge(:dump => ThreadsReport.generate(filter2)))
   end
```

Comment:
Happy to incorporate any suggestions for a more elegant way of doing this in Ruby.

Reply:
what are we attempting to accomplish, exactly?

just stringifying the keys and values in `filter`? If so, why don't we just build `filter` as needed in the line above?

~~~ ruby
filter = {'stacktrace_size' => "#{options.fetch(:stacktrace_size, HOT_THREADS_STACK_TRACES_SIZE_DEFAULT}" }
~~~

Reply:
Yes, thank you!